.github/workflows/build.yml

```
@@ -19,4 +19,84 @@
+        headers=(
+          -H "Authorization: Basic $patEncoded"
+          -H "Content-Type: application/json"
+        )
+
+        # Get the pool ID
+        url="https://dev.azure.com/${organization}/_apis/distributedtask/pools?poolName=${poolName}&api-version=7.1"
+        AZP_POOL_AGENTS=$(curl -s "${headers[@]}" -X GET "$url")
+        poolId=$(echo "$AZP_POOL_AGENTS" | jq -r '.value[0].id')
+
+        echo "Pool ID: $poolId"
+        
```

yamllint

[error] trailing spaces

(trailing-spaces)